src/cargo/util/toml/mod.rs

```diff
@@ -1611,6 +1611,18 @@ impl DetailedTomlDependency {
             }
         }
 
+        if let Some(git) = self.git.clone() {
```

Comment:
I think it would be simpler to move this down into the section where the git dependency is handled, just after the line `let loc = git.into_url()?;`.  That way, there's no need to convert it to a URL twice, and would remove 4 lines of code.

Reply:
Good idea, will do that